### PR TITLE
Be able to support a unified build in CMake.

### PR DESCRIPTION
**Pull request by mikeurbach**

### CODE REVIEW

**Comment by mortbopet** [comment]
Locally, i'm having the same issues as those thrown in the CI job https://github.com/circt-hls/circt-hls/runs/4792838341?check_suite_focus=true.

File: CMakeLists.txt [hunk 1]
```diff
@@ -13,34 +11,52 @@ endif()
...
-message(STATUS "Using LLVMConfig.cmake in: ${LLVM_DIR}")
+  message(STATUS "Using CIRCTConfig.cmake in: ${CIRCT_DIR}")
+  message(STATUS "Using MLIRConfig.cmake in: ${MLIR_DIR}")
+  message(STATUS "Using LLVMConfig.cmake in: ${LLVM_DIR}")
 
-set(LLVM_RUNTIME_OUTPUT_INTDIR ${CMAKE_BINARY_DIR}/bin)
-set(LLVM_LIBRARY_OUTPUT_INTDIR ${CMAKE_BINARY_DIR}/lib)
+  set(LLVM_RUNTIME_OUTPUT_INTDIR ${CMAKE_BINARY_DIR}/bin)
+  set(LLVM_LIBRARY_OUTPUT_INTDIR ${CMAKE_BINARY_DIR}/lib)
+  set(CIRCT_TOOLS_DIR ${CMAKE_BINARY_DIR}/bin/)
```
Comment (mikeurbach) [inline CMakeLists.txt]: I'm not sure how to set this for a non-unified build, and it appears to be failing CI in main as well. I tried fixing it with this line in this PR, but I'm not sure what to set it to. `CIRCT_BINARY_DIR` doesn't appear to be set in this kind of build.

**Comment by mikeurbach** [comment]
@mortbopet can you take another look when you get a chance? I think I've fixed the previous issue, but I'm not sure how to set CIRCT_TOOLS_DIR, which doesn't seem to be specific to this PR.

**Comment by mortbopet** [comment]
Rebasing removed the dependencies on the CIRCT cmake functions (`add_circt_dialect`...) in favor of just using the MLIR ones. This allows us to remove the `AddCIRCT` call. Things seem to be working fine on my end now!

**Pull request merged**